### ISSUE TITLE
Statistiken?

### ISSUE THREAD

**Issue by chrismarquardt** [eazyliving/firtz#104]
Welche Möglichkeiten habe ich beim firtz, an Stats für Feedzugriffe und Downloads zu kommen?

Feedburner oder händisches Logfile-fleddern? Oder vielleicht beim nächsten firtz doch was als Extension?

**Comment by eazyliving** [comment]
Hmhmh...

Hmhmhmhmhmhmhm...

:)

Ja, dem sollte eigentlich nicht mehr viel im Wege stehen. Doch auch wenn die Stochastik mir praktisch den Arsch im Abi gerettet hat (ich war sonst eine echte Pfeife), sind Statistiken etwas, das ich nicht gut kann :)

Das Extensionsystem sollte an der Stelle eigentlich gute Dienste leisten. Schlecht machbar werden jedoch solche Dinge wie Kampagnen sein, schätze ich. Aber das wäre zu prüfen.

Die grundsätzlichen Statistiken für Webseite und Feed sollten jedoch machbar sein. Interessant wird es für einen Podcast natürlich bei den Downloadstatistiken. Da wird ein Hook, eine Umleitung nötig sein, denn die Downloads gehen ja komplett am firtz vorbei irgendwo hin auf ein CDN oder sonstwas.

Das ist ein ziemlich gewaltiges Projekt. Ich habe mir die Statistiken von Podlove noch nicht angeschaut, aber sie werden überall schwer gelobt. Vielleicht ist es machbar, von dort Code zu entleihen, wenigstens den für die Darstellung.

Was auch ein nicht zu unterschätzendes Problem ist, ist die Datenbank. Der firtz hat keine. Und das bleibt so. Nur über meine Leiche lässt sich das ändern :) Das Plugin müsste sich also auf jeden Fall auch darum kümmern. Vielleicht ist sqlite eine Alternative. Sollten die Statistiken eher auf niedrigem Level und ohne komplexe Abfragen daher kommen, wäre eine reine textbasierte (csv) Variante denkbar.

Wo wir übrigens an einem Punkt ankommen, der etwas schwierig wird. Der firtz ist ja komplett frei von Authentifikation. Die Statistiken sollten wiederum geschützt darstellbar sein. Im Grunde müsste zumindest der darstellende Teil eine eigene Seite sein, die geschützt außerhalb des firtz rumliegt.

Fragen über Fragen, aber sicher ein geiles Projekt, das durchführbar ist. Nur nicht von mir, muss ich gestehen. Ich kann gerne zur Stelle sein, wo der firtz gefragt ist und ggf. modifiziert werden muss. Das wird sicher passieren müssen an ein oder zwei Stellen des Extensionsystems. Aber gerade im Statistikteil bin ich keine Hilfe, ganz bestimmt :)

Was aber jetzt schon ginge, denke ich: Piwik bzw. google analytics. Wenn ich mal ein wenig Luft habe, sollte der 2er durch sein, werde ich mal ein Modul für piwik und eins für GA bauen.

Wow. So viel habe ich hier vermutlich noch nie geschrieben :)

**Comment by chrismarquardt** [comment]
Och, Du meinst, das geht nicht mal eben so schnell? :) Ja ja, ich sehe die
diversen Stolpersteine sehr wohl.

Firtz hat auf der einen Seite diesen unverschämt guten
schieb-nach-auphonic-und-der-rest-ist-automatisch-Charme. Dafür ist es dann
halt ab dem Zeitpunkt weitgehend ein Blindflug und das tut mir mittlerweile
weh genug, dass ich trotz zwei superbequemen firtzen tatsächlich am
überlegen bin, vielleicht doch wieder wo anders aufzuschlagen.

Es wäre zwar toll, wenn der Firtz die Stats ganz von alleine könnte, aber
mir ist natürlich schon bewusst, was für eine Aufgabe das wäre. Darum hier
ein Vorschlag für eine sicher viel einfacher zu implementierende
Diät-Version:

Hooks für externe Trackingsysteme einbauen. Feedburner (jaja, keiner weiß,
wie lange es das noch gibt) braucht nur einen redirect für die Feed-URL und
Podtrac (kost nix) funktioniert über einen Redirect, für den man nur einen
URL-Prefix hinterlegen können müsste. Format:
http://www.podtrac.com/pts/redirect.aac/<URI-der-aac-datei> (geht
selbstverständlich auch mit mp3, ogg, etc.) - Beispiel:
http://www.podtrac.com/pts/redirect.aac/www.podcast.de/medien/blah.aac wird
dann wieder zurück geschubst auf http://www.podcast.de/medien/blah.aac



2015-04-05 23:27 GMT+02:00 Christian Bednarek <notifications@github.com>:

**Comment by eazyliving** [comment]
Hm. Die Podtrac-Geschichte kann ich Dir vermutlich in ein paar Minuten zusammenschnitzen. Feedburner wiederum... Nunja: Das liegt ja in Deiner Hand. Fütter den Feed-URL des firtz an feedburner und gut :)

**Comment by chrismarquardt** [comment]
Öhnee, so war das nicht gemeint.. ich werde nicht die Clients meiner Hörer
mit einer Feedburner-URL füttern, die dann irgendwann mal plötzlich, wenn
Google sich das anders überlegt, nicht mehr geht  und dann sind meine Hörer
gleich mit futsch. Dazu mache ich das schon zu lange und habe auch schon
genügend entsprechende Horrorstories gehört.

Weitgehende Kontrolle über den eigenen Feed ist für mich oberstes Gebot.

Ich stelle mir das mit der Feedburner-Einbindung so vor:

1. der Hörer abonniert den ganz normalen firtz-Feed über
http://blahblah.de/podcast/m4a

2. das ist aber in Wahrheit (im firtz konfigurierbar) ein temp Redirect
nach Feedburner, bzw. an eine beliebige URL

Sollte Feedburner von Google gekillt werden, wird das einfach wieder
ent-konfiguriert oder auf einen anderen Trackingdienst umkonfiguriert.

So wird man als Podcaster nicht vom externen Feeddienst abhängig. So
schläft sich's besser.

Und ja, mir ist die Problematik mit den temp redirects bewusst, nicht alle
Clients respektieren das und speichern es unter Umständen als permanenten
Redirect ab. Trotzdem wäre das immer noch besser als ein Komplettverlust im
Schadensfall.



2015-04-06 0:09 GMT+02:00 Christian Bednarek <notifications@github.com>:

**Comment by eazyliving** [comment]
Hmhmh... Das ist aber ein Problem. Der firtz muss hier einen feed erzeugen, den er feedburner liefert. Aber gleichzeitig muss er den clients sagen, dass der feed woanders ist. Das ist in der Tat ein klein wenig komplizierter und vor allem geht's aktuell auch nicht :-}

Aber podtrac geht, solltest Du das nutzen... die extension hab ich gerade fertig geschrieben (in einer ersten, aber funktionierenden Version :))

**Comment by chrismarquardt** [comment]
Ja. In dem Fall müsste der firtz entweder den echten Feed über eine
Geheim-URL rausgeben, die dann entsprechend in Feedburner als Input
konfiguriert wird, oder noch einfacher: der firtz wird so schlau, dass er
Feedburner am User Agent erkennt (meines Wissens "FeedBurner/1.0") und in
dem Fall nicht redirected, sondern den Feed ausspuckt.

So, podtrac kann er schon? In der neuen 2.0er beta? Oder noch im alten code
(der bei mir im Einsatz ist)?

2015-04-06 0:57 GMT+02:00 Christian Bednarek <notifications@github.com>:

**Comment by eazyliving** [comment]
Nein, das hab ich so nur für den neuen firtz gebaut. Ich überlege gerade, ob der 1.4er das schon könnte mit den extensions... Mal nachschauen :)

Ja, das wäre sicher denkbar mit dem feedburner. Aber nur mit massivem Eingriff in die Codebase, nicht mit einer Extension. Und die Zeit, das zu bauen investiere ich gerade lieber in die 2, sorry.

**Comment by eazyliving** [comment]
Ha! Gute Nachrichten. Die podtrac-Extension klappt auch mit der 1.4.

**Comment by chrismarquardt** [comment]
Heididei! Podtrac jetzt quasi implementiert auf http://glutenfrei-kochen.de/podcast und http://abzugfm.de/ - muss nur noch warten, bis die Feeds im Podtrac-System beide komplett drin sind, dann wird's rundum freigeschaltet.

Heißen Dank, das nimmt mir schon mal gut Druck weg. Downloadzahlen sind ja eh wichtiger als Feed-Subscriber. Und in der 2.0 reicht mir das locker.

**Comment by eazyliving** [comment]
Hm. Der redir bei glutenfrei ist bei m4a nicht redir.aac sondern redir.m4a. Weshalb das so ist: keine Ahnung. Bei meine alten testfirtz sieht's so gut aus (http://test.firtz.org). 

abzugfm wiederum scheint die extension noch nicht zu haben, oder? die links jedenfalls sind die originalen.

**Comment by chrismarquardt** [comment]
Achso, vergessen: aac ging nicht, hab's in der extension auf m4a geändert.
Muss wohl identisch mit der File-Extension sein.

Auf abzugfm.de mache ich es morgen rein. Der macht erst dann sauber den
redirect, wenn der Feed richtig bei denen im System steht. Ansonsten
bekommst du ein paar gnaden-redirects und danach macht er gegen Missbrauch
dicht.

2015-04-06 23:24 GMT+02:00 Christian Bednarek <notifications@github.com>:

**Comment by eazyliving** [comment]
Ah! Ok, das wusste ich nicht. Ich hab keine Infos gefunden, wie podtrac funktioniert und hab mir einfach Deine Infos genommen. Ich ändere das an der extension.

**Comment by i42n** [comment]
Ich persönlich nutze dafür einfach die Stats von podseed. Grundsätzlich denke ich nicht, dass es sinnvoll ist, wenn der Firtz selbst so eine Ansicht einbaut. Tracking-URLs macht Sinn, aber alles was eine DB braucht, sollte nicht rein.

**Comment by eazyliving** [comment]
Es spricht so gar nichts dagegen, das als Extension anzubieten. Ist nur ein klein wenig komplexer, als mal eben... Und man muss wissen was man tut. Und wer weiß das schon? ;)

**Comment by eazyliving** [comment]
Ich werde das Thema mal bei Gelegenheit angehen. Geplant ist eine sqlite-DB, die im extensions-Ordner liegen wird und dort alle Daten entgegen nimmt. Mal sehen, wie das klappen kann. Vermutlich werden ein paar Hooks nötig.

Was ich nicht gut kann: Visualisieren. Ganz schlecht sogar. Also wird's anfangs sicher nur ein paar Zahlen zu lesen geben.

Wenn es hier jemanden gibt, der solchen Kram gut kann: Bitte melden :)

**Comment by McCouman** [comment]
?
